### PR TITLE
feat(pytest-benchmark): implement pytest-benchmark support

### PR DESCRIPTION
**Pull request by ericlaz**
This PR adds support for `pytest-benchmark`, [a pytest fixture for benchmarking code](https://pypi.org/project/pytest-benchmark/). It adds 14 metrics to the test span including `min`, `median`, `mean`, `max`, among others.

## Checklist

- [X] Change(s) are motivated and described in the PR description.
- [X] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/contrib/pytest_benchmark/plugin.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+
+def pytest_configure(config):
+    if config.pluginmanager.hasplugin("benchmark"):
+        config.pluginmanager.register(_PytestBenchmarkPlugin(), "_datadog-pytest-benchmark")
+
+
+class _PytestBenchmarkPlugin:
+    def __init__(self):
+        pass
```
Comment (emmettbutler) [inline ddtrace/contrib/pytest_benchmark/plugin.py:17]:
```suggestion
```

**Review by emmettbutler — CHANGES_REQUESTED** [review]
Thanks for the contribution! I made a few suggestions aimed at cleaning up the code and getting CI to pass. Overall I think this approach makes sense.

File: releasenotes/notes/add-pytest-benchmark-support-71c6505ca4b202f3.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    CI Visibility: added pytest-benchmark support for ddtracer
```
[the remark this answers is not in the document]
Comment (ericlaz) [reply releasenotes/notes/add-pytest-benchmark-support-71c6505ca4b202f3.yaml:4]: Thanks for the review, will commit your changes!

File: riotfile.py [hunk 3]
```diff
@@ -1806,6 +1806,28 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
                 ),
             ],
         ),
+        Venv(
```
[the remark this answers is not in the document]
Comment (ericlaz) [reply riotfile.py:1809]: Should be added now, thank you!

File: tests/contrib/pytest_benchmark/test_pytest_benchmark.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        assert spans[0].get_metric(STATISTICS_HD15IQR) is not None
+        assert spans[0].get_metric(STATISTICS_IQR) is not None
+        assert spans[0].get_metric(STATISTICS_IQR_OUTLIERS) is not None
+        assert spans[0].get_metric(STATISTICS_LD15IQR) is not None
+        assert spans[0].get_metric(STATISTICS_MAX) > 2
+        assert spans[0].get_metric(STATISTICS_MEAN) > 2
+        assert spans[0].get_metric(STATISTICS_MEDIAN) > 2
+        assert spans[0].get_metric(STATISTICS_MIN) > 2
+        assert spans[0].get_metric(STATISTICS_OPS) > 0
+        assert spans[0].get_tag(STATISTICS_OUTLIERS) is not None
```
Comment (romainkomorndatadog) [inline tests/contrib/pytest_benchmark/test_pytest_benchmark.py:92]: Should this at least check the type of the data, to validate that `set_tag_str()` was called?

File: riotfile.py [hunk 5]
```diff
@@ -1665,6 +1665,28 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
...
+            name="pytest-benchmark",
+            command="pytest {cmdargs} tests/contrib/pytest_benchmark/",
+            pkgs={"msgpack": latest},
+            venvs=[
+                Venv(
+                    pkgs={
+                        "more_itertools": "<8.11.0",
```
Comment (romainkomorndatadog) [inline riotfile.py:1675]: Is this a `pytest-benchmark` dependency? I don't see the plugin or tests using it.

File: tests/contrib/pytest_benchmark/test_pytest_benchmark.py (hunk 4 above)
Comment (ericlaz) [reply tests/contrib/pytest_benchmark/test_pytest_benchmark.py:112]: Good catch! This has been addressed now

**Review by emmettbutler — APPROVED** [review]

**Review by majorgreys — DISMISSED** [review]

**Review by romainkomorndatadog — APPROVED** [review]

**Review by majorgreys — APPROVED** [review]

**Pull request merged**